tests/kernel/stack/stack/src/main.c

```diff
@@ -125,7 +125,7 @@ static void thread_entry_fn_single(void *p1, void *p2, void *p3)
 	for (i = STACK_LEN; i; i--) {
 		k_stack_pop((struct k_stack *)p1, &tmp[i - 1], K_NO_WAIT);
 	}
-	zassert_false(memcmp(tmp, data1, STACK_LEN),
+	zassert_false(memcmp(tmp, data1, STACK_LEN * sizeof(stack_data_t)),
```

Comment:
What about `sizeof(tmp)` so that we don't depend on size of that array components?
I.e. what if `tmp` becomes of size `STACK_LEN + 1` or type of stored elements is `stack_SUPER_data_t`?
Let's implement a robust solution once we spotted that problem now.

Reply:
or, introduce another macro instead of STACK_LEN, so it is only defined once.